Week4/RecyclerView/app/src/main/java/com/janinculhaoglu/recyclerview/TVShowsAdapter.kt

```diff
@@ -0,0 +1,69 @@
+package com.janinculhaoglu.recyclerview
+
+import android.content.Context
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.recyclerview.widget.RecyclerView
+import kotlinx.android.synthetic.main.cell_tvshow.view.*
+
+
+class TVShowsAdapter(val clickListener: TvShowListClickListener) : RecyclerView.Adapter<TVShowsAdapter.TVShowsViewHolder>() {
+
```

Comment:
You always want to have the data you are working with in your adapter. The best practice is to create a method like this: 
```kotlin
fun setData(shows: List<TvShow>) {
this.shows.clear()
this.shows.addAll(shows)
notifyDataSetChanged()
}
```

Then you just use that method to bring your data to the adapter.